knative/eventing
# Provide volume with OIDC token in SinkBinding

Fixes #7323

## Proposed Changes

- :gift: Provide OIDC token in SinkBinding as mountpoint under `/oidc/token`

### Pre-review Checklist

- [ ] **At least 80% unit test coverage**
- [ ] **E2E tests** for any new behavior
- [ ] **Docs PR** for any user-facing impact
- [ ] **Spec PR** for any new API feature
- [ ] **Conformance test** for any change to the spec

**How to verify change**
* Install a broker: `kn broker create my-broker --class MTChannelBasedBroker`
* Install a container source (will add a sinkbinding under the hood and makes it easy to verify the binding through its deployment): `kn source container create my-container-source --image gcr.io/knative-nightly/knative.dev/eventing/cmd/heartbeats:latest --env POD_NAME=pod-name --env POD_NAMESPACE=default --sink broker:my-broker`
* Verify the container source deplyment has the volume with the token: `kubectl get deploy my-container-source-deployment -o yaml`
* Additionally you can check if the mounted secret contains a JWT: `kubectl get secret oidc-token-my-container-source-sinkbinding -ojsonpath='{.data.token}' | base64 -d`

**Release Note**
```release-note
Provide OIDC token in SinkBinding under /oidc/token path.
```
